tree/dataframe/inc/ROOT/RDFInterface.hxx

```diff
@@ -1347,6 +1346,28 @@ public:
       return allColumns;
    }
 
+   /////////////////////////////////////////////////////////////////////////////
+   /// \brief Return the type of a given column as a string.
+   ///
+   /// This is not an action nor a transformation, just a query to the RDataFrame object.
+   std::string GetColumnType(std::string_view column)
+   {
+      auto lm = GetLoopManager();
+      const auto &customCols = lm->GetCustomColumnNames();
+      const bool convertVector2RVec = true;
+      const auto isCustom = std::find(customCols.begin(), customCols.end(), column) != customCols.end();
+      if (!isCustom) {
+         return RDFInternal::ColumnName2ColumnTypeName(std::string(column), lm->GetID(), lm->GetTree(),
+                                                       lm->GetDataSource(), isCustom, convertVector2RVec);
+      } else {
+         // must convert the alias "__tdf::column_type" to a readable type
+         const auto call = "ROOT::Internal::RDF::TypeID2TypeName(typeid(__tdf" + std::to_string(lm->GetID()) +
+                           "::" + std::string(column) + "_type))";
```

Comment:
I like the idea of this function in general. I just ask myself, though: instead of making two copies of the string_view you take as input, why not just take a const ref to string?

Reply:
for a utility like this i prioritized usability over performance, and `std::string_view` also accepts `TString`. consistency is another argument: all (most?) of RDF interfaces take string_views instead of const string refs.

of course the downside is that things might get a bit awkward in the function body, like here.

i don't have a strong opinion, if people vote to change it, i'll change it